pkg/engine/internal/planner/logical/planner.go

```diff
@@ -174,6 +197,11 @@ func buildPlanForLogQuery(
 		builder = builder.Select(value)
 	}
 
+	// TODO(chaudum): Drop stages can happen throughout the pipeline
```

Comment:
hmm, should we build the plan while traversing the AST to preserve the order? Stas made this change i believe, we can fix this in a follow-up